pkg/services/sqlstore/sqlstore_test.go

```diff
@@ -66,44 +68,50 @@ var sqlStoreTestCases = []sqlStoreTest{
 		dbHost:        "[::1]",
 		connStrValues: []string{"host=::1", "port=5432"},
 	},
+	{
+		name:  "Invalid database URL",
+		dbURL: "://invalid.com/",
+		err:   errors.New("parse \"://invalid.com/\": missing protocol scheme"),
+	},
 }
 
 func TestSQLConnectionString(t *testing.T) {
-	Convey("Testing SQL Connection Strings", t, func() {
-		t.Helper()
-
-		for _, testCase := range sqlStoreTestCases {
-			Convey(testCase.name, func() {
-				sqlstore := &SQLStore{}
-				sqlstore.Cfg = makeSQLStoreTestConfig(testCase.dbType, testCase.dbHost)
-				sqlstore.readConfig()
+	for _, testCase := range sqlStoreTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			sqlstore := &SQLStore{}
+			sqlstore.Cfg = makeSQLStoreTestConfig(t, testCase.dbType, testCase.dbHost, testCase.dbURL)
+			if err := sqlstore.readConfig(); err != nil {
+				require.Equal(t, testCase.err.Error(), err.Error())
+				return
+			}
 
-				connStr, err := sqlstore.buildConnectionString()
+			connStr, err := sqlstore.buildConnectionString()
+			require.Equal(t, testCase.err, err)
 
-				So(err, ShouldBeNil)
-				for _, connSubStr := range testCase.connStrValues {
-					So(connStr, ShouldContainSubstring, connSubStr)
-				}
-			})
-		}
-	})
+			for _, connSubStr := range testCase.connStrValues {
+				require.Contains(t, connStr, connSubStr)
+			}
+		})
+	}
 }
 
-func makeSQLStoreTestConfig(dbType string, host string) *setting.Cfg {
+func makeSQLStoreTestConfig(t *testing.T, dbType, host, dbURL string) *setting.Cfg {
 	cfg := setting.NewCfg()
 
 	sec, err := cfg.Raw.NewSection("database")
-	So(err, ShouldBeNil)
+	require.Nil(t, err)
```

Comment:
Nit, but we should use `require.NoError` instead of `require.Nil` to check for absence of errors:

```suggestion
	require.NoError(t, err)
```